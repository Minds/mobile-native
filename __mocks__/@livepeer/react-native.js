// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
