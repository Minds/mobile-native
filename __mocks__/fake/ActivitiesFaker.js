function load(count, container) {
  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid: code,
      custom_data: false,
      custom_type: false,
      rowKey: 'something' + i,
      description: 'Congratulations! ',
      edited: '',
      guid: code,
      mature: false,
      time_created: '1522036284',
      ownerObj: {
        guid: '824853017709780997',
        type: 'user',
        subtype: false,
        time_created: '1522036284',
      },
      message: 'Message',
      title: 'TITLE',
      owner_guid: '824853017709780997',
      parent_guid: '838106762591510528',
      perma_url: false,
      thumbnail_src: false,
      type: 'activity',
      wire_totals: {
        tokens: 1000000000000000000,
      },
      containerObj: container,
    };
  });

  return {
    activities: activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa',
  };
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load };
}

/**
 * Generate a single fake entity
 * @param {object} containerObj optional entity container
 * @returns
 */
export function fakeOne(containerObj) {
  const resp = load(1, containerObj);
  return resp.activities[0];
}
