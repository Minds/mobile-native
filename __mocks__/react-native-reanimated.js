import { mockView } from 'react-native';
export default {
  Value: jest.fn(),
  event: jest.fn(),
  add: jest.fn(),
  eq: jest.fn(),
  set: jest.fn(),
  cond: jest.fn(),
  interpolate: jest.fn(),
  sub: jest.fn(),
  multiply: jest.fn(),
  sqrt: jest.fn(),
  max: jest.fn(),
  diff: jest.fn(),
  onChange: jest.fn(),
  View: mockView,
  Extrapolate: { CLAMP: jest.fn() },
  Clock: jest.fn(),
  greaterThan: jest.fn(),
  lessThan: jest.fn(),
  startClock: jest.fn(),
  stopClock: jest.fn(),
  clockRunning: jest.fn(),
  not: jest.fn(),
  or: jest.fn(),
  and: jest.fn(),
  spring: jest.fn(),
  decay: jest.fn(),
  defined: jest.fn(),
  call: jest.fn(),
  block: jest.fn(),
  abs: jest.fn(),
  greaterOrEq: jest.fn(),
  lessOrEq: jest.fn(),
  debug: jest.fn(),
  Transition: {
    Together: 'Together',
    Out: 'Out',
    In: 'In',
  },
};
