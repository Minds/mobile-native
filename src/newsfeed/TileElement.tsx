import { MotiView, useAnimationState } from 'moti';
import React, { PureComponent } from 'react';
import { TouchableOpacity } from 'react-native';

import SmartImage from '../common/components/SmartImage';
import mediaProxyUrl from '../common/helpers/media-proxy-url';
import ThemedStyles from '../styles/ThemedStyles';
import type ActivityModel from './ActivityModel';

type PropsType = {
  entity: ActivityModel;
  size: number;
  navigation: any;
};

const transitionConfig: any = {
  type: 'timing',
};

function Container({
  children,
  ready,
}: {
  children: React.ReactNode;
  ready: boolean;
}) {
  const animationState = useAnimationState({
    from: {
      opacity: 0,
      scale: 0.7,
    },
    shown: {
      opacity: 1,
      scale: 1,
    },
  });

  React.useEffect(() => {
    if (ready) {
      animationState.transitionTo('shown');
    }
  }, [ready, animationState]);

  return (
    <MotiView state={animationState} transition={transitionConfig}>
      {children}
    </MotiView>
  );
}

export default class TileElement extends PureComponent<
  PropsType,
  { ready: boolean; source: any }
> {
  style: any;
  containerStyle: any;
  /**
   * Navigate to view
   */
  _navToView = () => {
    if (this.props.navigation) {
      this.props.navigation.push('Activity', { entity: this.props.entity });
    }
  };

  constructor(props) {
    super(props);
    const source = this.props.entity.getThumbSource();
    const type = this.props.entity.custom_type || this.props.entity.subtype;
    if (type === 'video') {
      source.uri = mediaProxyUrl(source.uri, 300);
    }
    this.state = {
      source,
      ready: false,
    };
    this.style = { width: this.props.size, height: this.props.size };
    this.containerStyle = ThemedStyles.combine(
      {
        width: this.props.size,
        height: this.props.size,
      },
      'bgSecondaryBackground',
      'borderHair',
      'bcolorPrimaryBorder',
    );
  }

  onLoadEnd = () => {
    this.setState({ ready: true });
  };

  render() {
    return (
      <TouchableOpacity onPress={this._navToView} style={this.containerStyle}>
        <Container ready={this.state.ready}>
          <SmartImage
            source={this.state.source}
            style={this.style}
            onLoadEnd={this.onLoadEnd}
          />
        </Container>
      </TouchableOpacity>
    );
  }
}
