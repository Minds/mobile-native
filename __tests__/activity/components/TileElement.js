import 'react-native';
import React from 'react';
import { TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import TileElement from '../../../src/newsfeed/TileElement';
import sp from '~/services/serviceProvider';

jest.mock('~/services/serviceProvider');

// mock services
sp.mockService('styles');

jest.mock('react-native-reanimated', () =>
  require('react-native-reanimated/mock'),
);

describe('Owner component', () => {
  let screen, navigation;
  beforeEach(() => {
    navigation = { navigate: jest.fn(), push: jest.fn() };

    let activityResponse = activitiesServiceFaker().load(1);
    let entity = activityResponse.activities[0];
    entity.getThumbSource = jest.fn();
    entity.getThumbSource.mockReturnValue({
      uri: 'https://www.minds.com/image',
    });

    screen = shallow(
      <TileElement entity={entity} navigation={navigation} size={30} />,
    );
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have Touchableopacity', async () => {
    screen.update();
    expect(screen.find(TouchableOpacity)).toHaveLength(1);
  });

  it('should _navToChannel on press ', () => {
    let activityResponse = activitiesServiceFaker().load(1);
    let entity = activityResponse.activities[0];
    entity.getThumbSource = jest.fn();
    entity.getThumbSource.mockReturnValue({
      uri: 'https://www.minds.com/image',
    });
    screen = shallow(
      <TileElement entity={entity} navigation={navigation} size={30} />,
    );
    screen.update();
    let touchables = screen.find('TouchableOpacity');
    touchables.at(0).props().onPress();

    expect(navigation.push).toHaveBeenCalledWith('Activity', {
      entity: entity,
    });
  });
});
