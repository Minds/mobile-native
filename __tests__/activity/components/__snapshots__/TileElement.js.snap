// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Owner component renders correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Object {
      "height": 30,
      "width": 30,
    }
  }
>
  <FastImage
    resizeMode="cover"
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  />
</TouchableOpacity>
`;
